service/frontend/dcRedirectionHandler.go

```diff
@@ -30,23 +35,59 @@ import (
 type (
 	// DCRedirectionHandlerImpl is simple wrapper over fontend service, doing redirection based on policy
 	DCRedirectionHandlerImpl struct {
-		currentClusteName string
-		redirectionPolicy DCRedirectionPolicy
-		tokenSerializer   common.TaskTokenSerializer
-		sevice            service.Service
-		frontendHandler   *WorkflowHandler
+		currentClusterName string
+		domainCache        cache.DomainCache
+		config             *Config
+		redirectionPolicy  DCRedirectionPolicy
+		tokenSerializer    common.TaskTokenSerializer
+		service            service.Service
+		frontendHandler    *WorkflowHandler
 	}
 )
 
+const (
+	// DomainActivenessAttemptCount indicate the attempts for the following API
+	// if seeing DomainNotActiveError
+	//
+	// 1. StartWorkflowExecution
+	// 2. SignalWithStartWorkflowExecution
+	// 3. SignalWorkflowExecution
+	DomainActivenessAttemptCount = 2
+
+	// APIAttemptCount is the attempt for normal API calls
+	APIAttemptCount = 1
+)
+
+// NewDCRedirectionHandlerWithWorkflowHandler create a dc redirection handler with existing frontend handler
+func NewDCRedirectionHandlerWithWorkflowHandler(wfHandler *WorkflowHandler, policy config.DCRedirectionPolicy) *DCRedirectionHandlerImpl {
```

Comment:
naming nit: both this and other method take workflowHandler as input - I don't see why this is called WithWorkflowHandler ? Consider renaming